File: Java Solutions/PermutedArithmeticSequence.java
import java.util.Arrays;
import java.util.Scanner;

public class PermutedArithmeticSequence {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int cases = sc.nextInt();

		while (cases != 0) {

			int nums = sc.nextInt();

			int arr[] = new int[nums];

			for (int i = 0; i < arr.length; i++) {
				arr[i] = sc.nextInt();
			}

			int count = 0;

			for (int k = 0; k < arr.length - 2; k++) {
				if (arr[arr.length - 1 - k] - arr[arr.length - 2 - k] == arr[arr.length - 2 - k]
						- arr[arr.length - 3 - k]) {
					count++;
				}
			}

			if (count == arr.length - 2) {
				System.out.println("arithmetic");
			} else {
				count = 0;
				Arrays.sort(arr);
				for (int k = 0; k < arr.length - 2; k++) {
					if (arr[arr.length - 1 - k] - arr[arr.length - 2 - k] == arr[arr.length - 2 - k]
							- arr[arr.length - 3 - k]) {
						count++;
					}
				}

				if (count == arr.length - 2) {
					System.out.println("permuted arithmetic");
				} else {
					System.out.println("non-arithmetic");

				}
			}

			count = 0;
			cases--;
		}

	}
}

File: Kattis-Problems/Python-Solutions/PermutedArithmeticSequence.py
a = int(input())

for k in range(a):

    count = 0
    arr = list(map(int, input().split()))
    numArr = arr[0]
    arr.remove(arr[0])

    for i in range(numArr-2):
        if arr[len(arr)-1-i] - arr[len(arr)-2-i] == arr[len(arr)-2-i]- arr[len(arr)-3-i]:
            count += 1

    if count == numArr-2:
        print('arithmetic')
        continue
    else:
        count = 0
        arr.sort()

        for j in range(numArr-2):
            if arr[len(arr)-1-j] - arr[len(arr)-2-j] == arr[len(arr)-2-j] - arr[len(arr)-3-j]:
                count += 1

        if count == numArr-2:
            print('permuted arithmetic')
            continue
        else:
            print('non-arithmetic')
